web/src/components/ui/ui.tsx

```diff
@@ -91,10 +103,14 @@ const LabeledFormControl = React.forwardRef(
           props.disabled ? 'disabled' : '',
         ].join(' ')}
         {...props}>
-        <span className="label">
-          {required && '*'}
-          {label}
-        </span>
+        {isLabelVisuallyHidden ? (
```

Comment:
Here if the label is hidden we use the new `VisuallyHidden` component. Otherwise we render the label as expected. I've wrapped the * with `aria-hidden` as I don't want that to be read out my assisitive technology.